Sources/MapboxCoreNavigation/AlternativeRoutesCenter.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+/**
+ `AlternativeRoutesCenter` observer to track alternative routes updates.
+ */
+public protocol AlternativeRoutesObserving: AnyObject {
+    /**
+     Called when center has detected a change in alternative routes list.
+     
+     - parameter center: `AlternativeRoutesCenter` reporting an update
+     - parameter didReportNewAlternatives: Indicies within `AlternativeRoutesCenter.alternatives` array where new alternative routes are appended.
+     - parameter removedAlternatives: Array of alternative routes which are no longer actual.
+     */
+    func alternativeRoutesCenter(_ center: AlternativeRoutesCenter, didReportNewAlternatives: IndexSet, removedAlternatives: [AlternativeRoute])
+    
+    /**
+     Called when center has failed to  change alternative routes list.
+     
+     - parameter center: `AlternativeRoutesCenter` reporting an update
+     - parameter didFailToUpdateAlternatives: An error occured.
+     */
+    func alternativeRoutesCenter(_ center: AlternativeRoutesCenter, didFailToUpdateAlternatives: AlternativeRouteError)
+}
+
+/**
+ Provides notifications and access to `AlternativeRoute`s found during navigation.
+ */
+public class AlternativeRoutesCenter {
+    
+    var observers: [AlternativeRoutesObserving] = []
```

Comment:
Do we want to retain observers here?

Reply:
Probably not – that would lead to potential retain cycles, as I described in https://github.com/mapbox/mapbox-navigation-ios/pull/3833#discussion_r882890001.

Reply:
Reworked `AlternativeRoutesCenter` so that it does not have observers list at all. Instead this functionality was appended to the RouterDelegate and Notifications mechanism.